interp/interp_test.go

```diff
@@ -3427,19 +3426,55 @@ var testBuiltinsMap = map[string]func(HandlerContext, []string) error{
 		return os.Link(oldname, newname)
 	},
 	"touch": func(hc HandlerContext, args []string) error {
+		filenames := args // create all arugments as filenames
+
 		newTime := time.Now()
-		if args[0] == "-d" {
-			if !strings.HasPrefix(args[1], "@") {
-				return fmt.Errorf("unimplemented")
+		if args[0] == "-t" {
+			if len(args) < 3 {
+				return fmt.Errorf("usage: touch [-t [[CC]YY]MMDDhhmm[.SS]] file")
+			}
+			filenames = args[2:] // treat the rest of the args as filenames
+
+			arg := args[1]
+			if strings.HasPrefix(arg, "-") {
+				return fmt.Errorf("usage: touch [-t [[CC]YY]MMDDhhmm[.SS]] file")
+			}
+			var offset int // index after yy
+			var (          // optional date fields
+				yy    = "2022"
```

Comment:
we just need this builtin for the tests; I think we can just force the user to always include the year :)